src/ngx_pagespeed.cc

```diff
@@ -1988,9 +1981,22 @@ void ps_send_to_pagespeed(ngx_http_request_t* r,
                           ngx_chain_t* in) {
   ngx_chain_t* cur;
   int last_buf = 0;
+
   for (cur = in; cur != NULL; cur = cur->next) {
     last_buf = cur->buf->last_buf;
 
+    if (cur->buf->flush && ctx->follow_flushes) {
+      // Calling ctx->proxy_fetch->Flush(cfg_s->handler) will be a no-op here,
+      // unless we have follow_flushes or flush_html enabled. Note that PSOL
+      // might aggregate multiple flushes into 1, and actually flush a little bit
+      // later due to html parser state and earlier scheduled operations.
+      // Also, unless we also set the flush flag on the nginx buffers we won't
+      // actually flush.
```

Comment:
When `string_piece_to_buffer_chain` is called by  `NgxBaseFetch` it will always have the `flush` argument set to `true`.
By default, we will never flush at all except when we are done transforming (equalling current behaviour).
We could consider switching the new `follow_flushes` argument to true for nps, which would make us flush when the upstream indicates it want us to. We could recommend enabling `flush_html` for reverse proxy setups.